src/scripts/ci/ci_tlsanvil_test.py

```diff
@@ -0,0 +1,113 @@
+# Script to run inside the CI container to test the botan
+# TLS client/server with TLS-Anvil
+#
+# (C) 2023 Jack Lloyd
+# (C) 2023 Fabian Albert, Rohde & Schwarz Cybersecurity
+#
+# Botan is released under the Simplified BSD License (see license.txt)
+import sys
+import argparse
+import os
+import subprocess
+
+
+class Config:
+    """ Hardcoded configurations for this CI script """
+    key_and_cert_storage_path = "/tmp/"
+    test_suite_results_path = "./TestSuiteResults"
+    tmp_key_file_name = "tmp_rsa_key.pem"
+    tmp_cert_file_name = "tmp_rsa_cert.pem"
+    server_dest_ip = "127.0.0.1"
+    server_dest_port = 4433
+    botan_server_log = "./logs/botan_server.log"
+
+
+def create_cert_and_key(botan_exe_path):
+    """
+    Create a X.509 certificate and associated RSA key at Config.key_and_cert_storage_path
+    using Botan's CLI.
+
+    Returns: (<cert path>, <key path>)
+    """
+
+    key_path = os.path.join(Config.key_and_cert_storage_path, Config.tmp_key_file_name)
+    cert_path = os.path.join(Config.key_and_cert_storage_path, Config.tmp_cert_file_name)
+
+    with open(key_path, 'w', encoding='utf-8') as keyfile:
+        subprocess.run([botan_exe_path, "keygen", "--algo=RSA"], stdout=keyfile, check=True)
+
+    with open(cert_path, 'w', encoding='utf-8') as certfile:
+        subprocess.run([botan_exe_path, "gen_self_signed", key_path, "localhost"], stdout=certfile, check=True)
+
+    return (cert_path, key_path)
+
+
+def server_test(botan_exe_path: str, tls_anvil_jar_path: str):
+    cert_path, key_path = create_cert_and_key(botan_exe_path)
+
+    tls_anvil_cmd = [
+        "java", "-jar", tls_anvil_jar_path,
+        "-strength", "1",
+        "-parallelHandshakes", "1",
```

Comment:
The runners we get have 2 cores - would it make sense / be possible to use 2-3 handshakes in parallel here?

Reply:
Yeah, parallel handshakes would be great. However, I run TLS-Anvil against the `./botan tls_server` process, which only allows one connection at a time.

Reply:
Could we use `tls_http_server` instead?